### PR TITLE
Adding release-it configuration with workspaces publishing support

### PR DESCRIPTION
**Pull request by scalvert**
~Depends on #1~

Adds `release-it` and associated configuration for yarn workspaces via [`create-rwjblue-release-it-setup`](https://github.com/rwjblue/create-rwjblue-release-it-setup). 

Note: we're using the `release-it-yarn-workspaces` plugin even though we use `npm` in this repository. That plugin works with `npm`, and will ultimately be renamed to the more generic `release-it-workspaces` to correctly signify generic workspaces support.

### CODE REVIEW

**Comment by scalvert** [comment]
cc/ @jeffersonking

File: package.json [hunk 1]
```diff
@@ -1,12 +1,12 @@
 {
```
Comment (scalvert) [inline package.json:1]: The `create-rwjblue-release-it-setup` package uses `sort-package-json`, which is why this file now appears sorted in spite of minimal changes.

**Comment by scalvert** [comment]
Converting this to a draft for now while @eddynaka investigates requirements for publishing.

**Comment by eddynaka** [comment]
> Converting this to a draft for now while @eddynaka investigates requirements for publishing.

Quick update on this:
- started a pr with codeql
- integrated with LGTM

Internally people run eslint. Should we run as well?

**Comment by scalvert** [comment]
@eddynaka since our [github actions run eslint](https://github.com/microsoft/sarif-js-sdk/blob/main/.github/workflows/ci.yml#L25) code should not be able to be merged in a PR without that passing. We could additionally run lint checks before a release, but it does seem redundant. If we're concerned about folks pushing to main and circumventing this, we could make main a protected branch, which would require anyone committing anything to be forced down the PR/actions pipeline.

WDYT?

**Comment by eddynaka** [comment]
> @eddynaka since our [github actions run eslint](https://github.com/microsoft/sarif-js-sdk/blob/main/.github/workflows/ci.yml#L25) code should not be able to be merged in a PR without that passing. We could additionally run lint checks before a release, but it does seem redundant. If we're concerned about folks pushing to main and circumventing this, we could make main a protected branch, which would require anyone committing anything to be forced down the PR/actions pipeline.
> 
> WDYT?

no worries. I wasn't sure if we were already running. thanks for answering :)

**Comment by eddynaka** [comment]
@scalvert , I checked internally and everything that we set-up so far is good enough (CodeQL, LGTM, and ESLint).
We can continue this pr :)

**Comment by scalvert** [comment]
@eddynaka thanks for looking into it!

**Review by rwjblue — APPROVED** [review]

**Pull request merged**